File: csv_detective/formats/jour_de_la_semaine.py
proportion = 0.8
description = "Weekday name in French"
tags = ["fr", "temp"]
labels = {
    "jour semaine": 1,
    "type jour": 1,
    "jour de la semaine": 1,
    "saufjour": 1,
    "nomjour": 1,
    "jour": 0.75,
    "jour de fermeture": 1,
}

jours = {
    "lundi",
    "mardi",
    "mercredi",
    "jeudi",
    "vendredi",
    "samedi",
    "dimanche",
    "lun",
    "mar",
    "mer",
    "jeu",
    "ven",
    "sam",
    "dim",
}


def _is(val) -> bool:
    if not isinstance(val, str):
        return False
    return val.lower() in jours


_test_values = {
    True: ["lundi"],
    False: ["jour"],
}

File: rust/src/formats/jour_de_la_semaine.rs
use crate::value::Value;

const VALID: &[&str] = &[
    "lundi", "mardi", "mercredi", "jeudi", "vendredi", "samedi", "dimanche",
    "lun", "mar", "mer", "jeu", "ven", "sam", "dim",
];

pub fn detect(val: &str) -> Option<()> {
    let lower = val.to_lowercase();
    if VALID.contains(&lower.as_str()) {
        Some(())
    } else {
        None
    }
}

pub fn test(val: &Value) -> bool {
    detect(val.raw()).is_some()
}
